runtime/weak.js

```diff
@@ -114,21 +120,34 @@ function caml_ephe_get_key_copy(x, i) {
 }
 
 //Provides: caml_ephe_check_key mutable
-//Requires: caml_ephe_key_offset
+//Requires: caml_ephe_key_offset, caml_ephe_data_offset
+//Requires: caml_invalid_argument
+//Requires: caml_ephe_none
 //Alias: caml_weak_check
 function caml_ephe_check_key(x, i) {
+  if (i < 0 || caml_ephe_key_offset + i >= x.length)
```

Comment:
You can remove the checks. They have been moved to the OCaml code since OCaml 4.08 (https://github.com/ocaml/ocaml/pull/676).

Reply:
done